src/test/scala/com/snowflake/snowpark_test/SqlSuite.scala

```diff
@@ -104,6 +104,37 @@ trait SqlSuite extends SNTestBase {
       new Directory(new File(outputPath)).deleteRecursively()
     }
   }
+
+  test("Run sql query with bindings") {
+    val df1 = session.sql("select * from values (?),(?),(?)", List(1, 2, 3))
+    assert(df1.collect() sameElements Array[Row](Row(1), Row(2), Row(3)))
+
+    val df2 =
+      session.sql(
+        "select variance(identifier(?)) from values(1,1),(1,2),(2,1),(2,2),(3,1),(3,2) as T(a, b)",
+        Seq("a"))
+    assert(df2.collect()(0).getDecimal(0).toString == "0.800000")
+
+    val df3 = session
+      .sql("select * from values (?),(?),(?) as T(id)", Seq(1, 2, 3))
+      .filter(col("id") < 3)
+    assert(df3.collect() sameElements Array[Row](Row(1), Row(2)))
+
+    val df4 =
+      session.sql("select * from values (?,?),(?,?),(?,?) as T(a, b)", Seq(1, 1, 2, 1, 3, 1))
+    val df5 =
+      session.sql("select * from values (?,?),(?,?),(?,?) as T(a, b)", List(1, 2, 2, 1, 4, 3))
+    val df6 = df4.union(df5).filter(col("a") < 3)
+    assert(df6.collect() sameElements Array[Row](Row(1, 1), Row(2, 1), Row(1, 2)))
```

Comment:
it should have two `(2, 1)` right?

Reply:
![Screenshot 2024-11-01 at 11 12 53 AM](https://github.com/user-attachments/assets/4376516f-d5ae-4079-baf7-6993d40cecbd)

Apparently no for union (executed here), but yes for union all.